models/vrp.rb

```diff
@@ -337,49 +337,96 @@ def self.remove_unecessary_units(hash)
       hash[:shipments]&.map{ |s| (s[:quantities] || []).delete_if{ |quantity| needed_units.exclude? quantity[:unit_id] } }
     end
 
-    def self.generate_schedule_indices_from_date(hash)
-      return hash if !hash[:configuration] || !hash[:configuration][:schedule] ||
-                     hash[:configuration][:schedule][:range_indices]
+    def self.convert_availability_dates_into_indices(element, hash, start_index, type)
+      unavailable_indices, unavailable_dates =
+        case type
+        when :visit
+          [:unavailable_visit_day_indices, :unavailable_visit_day_date]
+        when :vehicle
+          [:unavailable_work_day_indices, :unavailable_work_date]
+        when :schedule
+          [:unavailable_indices, :unavailable_date]
+        end
 
-      start_indice = hash[:configuration][:schedule][:range_date][:start].to_date.cwday - 1
-      end_indice = (hash[:configuration][:schedule][:range_date][:end].to_date - hash[:configuration][:schedule][:range_date][:start].to_date).to_i + start_indice
+      element[unavailable_indices] ||= []
+      element[unavailable_dates].to_a.each{ |unavailable_date|
+        element[unavailable_indices] |=
+          [(unavailable_date.to_date - hash[:configuration][:schedule][:range_date][:start]).to_i + start_index]
```

Comment:
`hash[:configuration][:schedule][:range_date][:start].to_i + start_index` could be put within a variable as it is used 3 times.

Reply:
I am not sure if this is possible : only difference between two dates can be converted to_i, it seems to me that this value needs to be computed from each corresponding date.

Reply:
Indeed, `.to_i` has no meaning, as it requires a zero. There is multiple way to obtain an integer from a date.
The difference between two dates returns a rational. But you can obtain this rational with `.ajd` or you can get the integer using `.jd` (if you want it to) (julian calendar)

`.ld` in Lilian calendar